src/integrations/xiaomi-mi/xiaomi-mi.service.ts

```diff
@@ -0,0 +1,211 @@
+
+  constructor(
+    private readonly entitiesService: EntitiesService,
+    private readonly configService: ConfigService,
+    private readonly schedulerRegistry: SchedulerRegistry
+  ) {
+    this.config = this.configService.get('xiaomiMi');
+    this.logger = new Logger(XiaomiMiService.name);
+  }
+
+  /**
+   * Lifecycle hook, called once the host module has been initialized.
+   */
+  onModuleInit(): void {
+    if (!this.hasSensors()) {
+      this.logger.warn(
+        'The addresses are empty, no sensors will be created! Please add some of the discovered IDs below to your configuration.'
+      );
+    }
+  }
+
+  /**
+   * Lifecycle hook, called once the application has started.
+   */
+  onApplicationBootstrap(): void {
+    this.config.sensors.forEach(options => {
+      this.sensors[options.address] = new SensorSet(
+        options.name,
+        this.entitiesService
+      );
+    });
+    noble.on('stateChange', XiaomiMiService.handleStateChange);
+    noble.on('discover', this.handleDiscovery.bind(this));
+    noble.on('warning', this.onWarning.bind(this));
+    this.logger.log('XiaomiMi Service bootstrapped!');
+  }
+
+  /**
+   * Log warnings from noble.
+   */
+  onWarning(message: string): void {
+    this.logger.warn('Warning: ', message);
+  }
+
+  /**
+   * Determines whether we have any sensors.
+   *
+   * @returns Sensors status
+   */
+  hasSensors(): boolean {
+    return this.config.sensors?.length > 0;
+  }
+
+  /**
+   * Filters found BLE peripherals and publishes new readings to sensors, depending on configuration.
+   *
+   * @param peripheral - BLE peripheral
+   */
+  handleDiscovery(peripheral: Peripheral): void {
+    const { advertisement, id } = peripheral || {};
+    let sensorSet = this.sensors[id];
+    if (!sensorSet) {
+      return;
+    }
+    const buffer = this.getValidServiceData(advertisement);
+    if (!buffer) {
+      return;
+    }
+    const serviceData = this.parseServiceData(buffer);
+    if (!serviceData) {
+      return;
+    }
+    const event = serviceData.event;
+    switch (serviceData.eventType) {
+      case EventTypes.temperature: {
+        sensorSet.recordMeasure('temperature', '°C', event.temperature);
+        break;
+      }
+      case EventTypes.humidity: {
+        sensorSet.recordMeasure('humidity', '%', event.humidity);
+        break;
+      }
+      case EventTypes.battery: {
+        this.logger.log(`Got battery: ${event.battery}`);
+        sensorSet.recordMeasure('battery', '%', event.battery);
+        break;
+      }
+      case EventTypes.temperatureAndHumidity: {
+        this.logger.log(
+          `Got temperature: ${event.temperature} and humidity: ${event.humidity}`
+        );
+        sensorSet.recordMeasure('temperature', '°C', event.temperature);
+        sensorSet.recordMeasure('humidity', '%', event.humidity);
+        break;
+      }
+      case EventTypes.illuminance: {
+        this.logger.log(`Got illuminance: ${event.illuminance}`);
```

Comment:
Is there a reason why this event type and the ones below it are not recorded in entities and only logged?

Reply:
I don't have a device to test these event types.   But I fixed it to at least treat them consistently with others.